src/Features/Core/Portable/Diagnostics/Analyzers/PopulateSwitchDiagnosticAnalyzerBase.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using System.Collections.Immutable;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.Diagnostics.PopulateSwitch
+{
+    internal abstract class PopulateSwitchDiagnosticAnalyzerBase<TLanguageKindEnum> : DiagnosticAnalyzer, IBuiltInAnalyzer where TLanguageKindEnum : struct
+    {
+        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(FeaturesResources.AddMissingStatements), FeaturesResources.ResourceManager, typeof(FeaturesResources));
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(WorkspacesResources.PopulateSwitch), WorkspacesResources.ResourceManager, typeof(WorkspacesResources));
+
+        private static readonly DiagnosticDescriptor s_descriptor = new DiagnosticDescriptor(IDEDiagnosticIds.PopulateSwitchDiagnosticId,
+                                                                    s_localizableTitle,
+                                                                    s_localizableMessage,
+                                                                    DiagnosticCategory.EditAndContinue,
+                                                                    DiagnosticSeverity.Hidden,
+                                                                    isEnabledByDefault: true,
+                                                                    customTags: DiagnosticCustomTags.Unnecessary);
+
+        #region Interface methods
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+        {
+            get
+            {
+                return ImmutableArray.Create(s_descriptor);
+            }
+        }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(
```

Comment:
I feel like this shuld be an IOperation analyzer.  And you shoudl be registering to be called back on "ISwitchStatement"  This could then be written for VB and C# in a single place.